test/replication-luatest/gh_6966_readonly_bootstrap_test.lua

```diff
@@ -11,20 +11,19 @@ g.before_all(function(cg)
 
     local cfg = {
         replication_timeout = 0.1,
-        replication = server.build_listen_uri('master'),
+        replication = {
+            server.build_listen_uri('master'),
+        },
     }
-    -- XXX: the order of add_server() is important. First add replica, then
-    -- master. This way they are stopped by cluster:stop() in correct order,
-    -- and the test runs 3 seconds faster. See gh-6820 (comment) for details:
-    -- https://github.com/tarantool/tarantool/issues/6820#issuecomment-1082914500
-    cg.replica = cg.cluster:build_and_add_server({
-        alias = 'replica',
-        box_cfg = cfg
-    })
     cg.master = cg.cluster:build_and_add_server({
         alias = 'master',
         box_cfg = cfg
     })
+    cfg.replication[2] = server.build_listen_uri('replica')
+    cg.replica = cg.cluster:build_and_add_server({
+        alias = 'replica',
+        box_cfg = cfg
+    })
```

Comment:
Did you check that the test is not running 3 seconds longer now?

Reply:
It isn't:
```
======================================================================================
WORKR TEST                                            PARAMS          RESULT
---------------------------------------------------------------------------------
[001] replication-luatest/gh_6966_readonly_bootstrap>                 [ pass ]
---------------------------------------------------------------------------------
Top 10 tests by occupied memory (RSS, Mb):
*   29.9 replication-luatest/gh_6966_readonly_bootstrap_test.lua      

(Tests quicker than 0.1 seconds may be missed.)

---------------------------------------------------------------------------------
Top 10 longest tests (seconds):
*   0.57 replication-luatest/gh_6966_readonly_bootstrap_test.lua      
```
Besides, it would hang now, because we have `on_shutdown_trigger_timeout` set to infinity somewhere in test config defaults. To catch all such kinds of errors:
```
./test-run/lib/luatest/luatest/server_instance.lua:    box.ctl.set_on_shutdown_timeout(TIMEOUT_INFINITY)

```